src/encoder.rs

```diff
@@ -94,6 +119,30 @@ impl Encoder {
         }
         .check_result()
     }
+
+    /// Get the next page from the stream
+    ///
+    /// Only use if creating encoder with [`Encoder::create_pull`].
+    pub fn get_page(&mut self, flush: bool) -> Option<Vec<u8>> {
```

Comment:
I think we don't have to copy the page into a new `Vec<8>`. Looking at https://github.com/xiph/libopusenc/blob/e4285b5e8c20a06746f656102c5be8aef83f7ccb/src/opusenc.c#L808-L815 and https://github.com/xiph/libopusenc/blob/e4285b5e8c20a06746f656102c5be8aef83f7ccb/src/opusenc.c#L808-L815, the pointer is valid until the next call to `ope_encoder_get_page()`. Since we take `&mut self`, returning the slice directly should work and I think we don't even need explicit lifetimes.

```suggestion
    pub fn get_page(&mut self, flush: bool) -> Option<&[u8]> {
```